src/types.js

```diff
@@ -1,9 +1,19 @@
 import _ from 'lodash/fp'
 import * as F from 'futil-js'
 
-export let getTypeProp = _.curry((types, prop, node) =>
-  F.cascade([`${node.type}.${prop}`, `default.${prop}`], types)
+// Gets type a specific property from any of the places it might be - on the type in `types`, on default in `types`, or already on the node itself
+export let getTypeProp = _.curry(
```

Comment:
I think the opposite of this `OnError` naming convention is more widely used. The opposite being `maybe[name]` for the fault-tolerant function and `[name]` for the more strict function.